Chessington.GameEngine/Pieces/Knight.cs

```diff
@@ -6,11 +6,31 @@ namespace Chessington.GameEngine.Pieces
     public class Knight : Piece
     {
         public Knight(Player player)
-            : base(player) { }
+            : base(player)
+        {
+        }
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var availablePositions = new List<Square>();
+            var currentSquare = board.FindPiece(this);
+
+            var rowPos = new[] { -2, -2, 1, -1, 2, 2, 1, -1 };
+            var colPos = new[] { 1, -1, 2, 2, 1, -1, -2, -2 };
+
+            for (var at = 0; at < 8; at++)
```

Comment:
You have a `GameSettings.BoardSize` static variable that you can use instead of the magic number 8